x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/mapper/GeoShapeWithDocValuesFieldMapper.java

```diff
@@ -283,4 +283,13 @@ public GeoShapeWithDocValuesFieldType fieldType() {
         return (GeoShapeWithDocValuesFieldType) super.fieldType();
     }
 
+    @Override
+    protected void checkIncomingMergeType(FieldMapper mergeWith) {
+        if (mergeWith instanceof AbstractShapeGeometryFieldMapper<?>
+            && (mergeWith instanceof GeoShapeWithDocValuesFieldMapper) == false) {
```

Comment:
These.second set of parentheses around `(mergeWith instanceof GeoShapeWithDocValuesFieldMapper)` is a bit  confusing.

Reply:
I simplified the check . We cannot  place super.checkIncomingMergeType(mergeWith) because it will throw an error with a confusing error.